include/deal.II/grid/tria_iterator.h

```diff
@@ -1036,7 +1014,7 @@ template <typename Accessor>
 inline const Accessor *
 TriaRawIterator<Accessor>::operator->() const
 {
-  return &(this->operator*());
```

Comment:
!!! How did we never notice this? !!!

Reply:
Two reasons I think. First, deal.II has its own idiosynchrasies and users tend to follow them (for example our `cell->`, `face->` usage throughout the examples, etc.). Secondly, for a long time it wasn't easily possible to take a reference of just about anything. This only came with `auto`. And even though this is C++ 11, the actual era of `auto` in the library started much later.

Reply:
But this isn't wrong. In fact, you're replacing the code with something I consider worse. The previous syntax made clear that `operator->` and `operator*` are at the core the same thing: One is implemented in terms of the other.